### DISCUSSION TITLE
[2024-01-03] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8666] [Weekly Demo Day]
@farmaazon 

* Finished fixing dynamic configuration for widgets. It required more widgets refactoring, as the dynamic config may provide configuration quite deeply: https://github.com/enso-org/enso/pull/8624

[Screencast from 2024-01-03 15-55-57.webm](https://github.com/enso-org/enso/assets/3919101/04e704f9-0b7d-4f24-afa9-646a133abc0b)

There are still issues with Table nodes' widgets: constructors (like in Aggregate) have no arguments displayed: #8663 Placeholders does not have all widgets (already in progress): #8661

@MichaelMauderer 

* https://github.com/enso-org/enso/pull/8623

[Screencast from 21-12-23 14:52:20.webm](https://github.com/enso-org/enso/assets/1428930/c1703844-2b17-435b-b8ff-0ef4a858d875)

**Comment by vitvakatu** [comment]
- Working on https://github.com/enso-org/enso/issues/8067
- Working on https://github.com/enso-org/enso/issues/8532 (later taken over by @farmaazon)


https://github.com/enso-org/enso/assets/6566674/4a107509-1dc6-4647-b52c-85307d7d8dfc

**Comment by kazcw** [comment]
Improved AST APIs and integrated further with GUI: #8625

**Comment by JaroslavTulach** [comment]
- **7000** times faster `DataflowError` #8608
-  size of `Table.ir` down from **6MB to 3.5MB** #8635

**Comment by Frizi** [comment]
- Attempted widget update implementation using new AST updates, but had many problems. The current version ended up being implemented by Kaz, now merged (https://github.com/enso-org/enso/pull/8625). Still need to investigate some visual bugs.
- Working on reordering nodes in code on edge connection. Code modification is done, but having issues with committing the AST edits. #8494
